avoid shippable build errors on limited platforms

pabigot opened the pull request:
Several platforms fail to build several tests because they run out of memory.  Exclude them so we don't get false negatives from shippable on PRs that happen to trigger these tests.

Found with `sanitycheck --all`.

tyrantlizard77 reviewed (changes requested):
This isn't the right way to do this.
The right way to do this is to set the platform's RAM/flash size in the board level YAML file, and then set min_ram/min_flash in the testcase.yaml.

pabigot commented:
> The right way to do this is to set the platform's RAM/flash size in the board level YAML file, and then set min_ram/min_flash in the testcase.yaml.

Fair enough.  In that case I'll have to increase the amount of flash required for some tests, because these boards are failing even though they meet the minimum specified.

I'll rework for that.

pabigot committed:
tests: exclude platforms with limited flash

Platforms with limited flash are now failing to link.  Add or increase
flash requirements for test cases to exclude the ones that will fail.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

tyrantlizard77 reviewed (approved):
looks better thanks.
I'm uneasy that basic kernel validation tests like thread_apis and poll use so much flash, but that can get addressed in another PR.
